test/e2e/storage/testsuites/provisioning.go

```diff
@@ -73,132 +76,88 @@ func (p *provisioningTestSuite) getTestSuiteInfo() TestSuiteInfo {
 	return p.tsInfo
 }
 
-func (p *provisioningTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver TestDriver) {
-}
-
-func createProvisioningTestInput(driver TestDriver, pattern testpatterns.TestPattern) (provisioningTestResource, provisioningTestInput) {
-	// Setup test resource for driver and testpattern
-	resource := provisioningTestResource{}
-	resource.setupResource(driver, pattern)
-
-	input := provisioningTestInput{
-		testCase: StorageClassTest{
-			ClaimSize:    resource.claimSize,
-			ExpectedSize: resource.claimSize,
-		},
-		cs:    driver.GetDriverInfo().Config.Framework.ClientSet,
-		pvc:   resource.pvc,
-		sc:    resource.sc,
-		dInfo: driver.GetDriverInfo(),
-	}
-
-	if driver.GetDriverInfo().Config.ClientNodeName != "" {
-		input.testCase.NodeName = driver.GetDriverInfo().Config.ClientNodeName
-	}
-
-	return resource, input
-}
-
-func (p *provisioningTestSuite) execTest(driver TestDriver, pattern testpatterns.TestPattern) {
-	Context(getTestNameStr(p, pattern), func() {
-		var (
-			resource     provisioningTestResource
-			input        provisioningTestInput
-			needsCleanup bool
-		)
-
-		BeforeEach(func() {
-			needsCleanup = false
-			// Skip unsupported tests to avoid unnecessary resource initialization
-			skipUnsupportedTest(p, driver, pattern)
-			needsCleanup = true
-
-			// Create test input
-			resource, input = createProvisioningTestInput(driver, pattern)
-		})
-
-		AfterEach(func() {
-			if needsCleanup {
-				resource.cleanupResource(driver, pattern)
-			}
-		})
-
-		// Ginkgo's "Global Shared Behaviors" require arguments for a shared function
-		// to be a single struct and to be passed as a pointer.
-		// Please see https://onsi.github.io/ginkgo/#global-shared-behaviors for details.
-		testProvisioning(&input)
-	})
-}
-
-type provisioningTestResource struct {
-	driver TestDriver
-
-	claimSize string
-	sc        *storage.StorageClass
-	pvc       *v1.PersistentVolumeClaim
-}
+func (p *provisioningTestSuite) defineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	var (
+		f             = framework.NewDefaultFramework("provisioning")
+		dInfo         = driver.GetDriverInfo()
+		config        *PerTestConfig
+		driverCleanup func()
+		testCase      *StorageClassTest
+		pvc           *v1.PersistentVolumeClaim
+		sc            *storage.StorageClass
+	)
+
+	init := func() {
```

Comment:
nit: `init` could be more descriptive. `initTestCase`?

Reply:
okay